apps/finance/app/src/script.js

```diff
@@ -169,16 +166,15 @@ async function createStore(settings) {
  *                     *
  ***********************/
 
-async function initializeState(state, settings) {
-  const nextState = {
-    ...state,
+const initializeState = settings => async () => {
+  const newState = {
     periodDuration: marshallDate(
       await app.call('getPeriodDuration').toPromise()
     ),
     vaultAddress: settings.vault.address,
   }
 
-  const withTestnetState = await loadTestnetState(nextState, settings)
+  const withTestnetState = await loadTestnetState(newState, settings)
   const withEthBalance = await loadEthBalance(withTestnetState, settings)
```

Comment:
⚠️ We should also refresh any existing token balances we know of here (could just loop through the balances and call `updateBalances()` on each token). If a user transfers some tokens to the Vault without using its `deposit()` functionality (by interacting with the token contract itself), and we don't process another event related to this token, we wouldn't know about it.